workflow-core/src/commonMain/kotlin/com/squareup/workflow1/WorkflowAction.kt

```diff
@@ -133,3 +143,28 @@ public class WorkflowOutput<out OutputT>(public val value: OutputT) : ActionProc
 
   override fun hashCode(): Int = value.hashCode()
 }
+
+/**
+ * An [ActionProcessingResult] is any possible outcome after the runtime does a loop of processing.
+ *
+ * Only [ActionApplied] needs the generic OutputT so we do not include it in the root
+ * interface here.
+ */
+public sealed interface ActionProcessingResult
```

Comment:
It's interesting that `ActionProcessingResult` was already public, never really thought about that. Wondering if it's actually necessary.

Reply:
It's used in runtime so yes.